### ISSUE TITLE
Feature Request: Alert once per environment per issue

### ISSUE THREAD

**Issue by cpressland** [getsentry/sentry#21612]
## Summary

An additional option to send alerts based on first time an issue is seen in a given environment instead of just the first time the issue is seen.

## Motivation

We use Microsoft Teams for alerting, we currently have the following setup in Sentry Alerts:

- "env: prod" > "The event occurs" > "Send to Teams Channel: 'Alerts - Production'"
- "env: staging" > "The event occurs" > "Send to Teams Channel: 'Alerts - Staging'"
- "env: dev" > "The event occurs" > "Send to Teams Channel: 'Alerts - Dev'"

We'd like to instead only fire alerts the first time an issue is seen within a specific environment, however, the following workflow doesn't work:

- "env: prod" > "The issue is first seen" > "Send to Teams Channel: 'Alerts - Production'"
- "env: staging" > "The issue is first seen" > "Send to Teams Channel: 'Alerts - Staging'"
- "env: dev" > "The issue is first seen" > "Send to Teams Channel: 'Alerts - Dev'"

With this setup we only ever see a single alert regardless of environment. If an issue first occurs in Staging, we'll never see a second alert and third alert in the Dev or Production channels. Instead, the alerting rules should give us the flexibility to alert once per env per issue.

**Comment by adhiraj** [comment]
Agreed this would be useful. In the meantime here's an approximate workaround:

- Select the environment
- In the "WHEN" clause don't select anything (which tells sentry to alert when a new event is seen)
- For rate limit select 30 days so that you get alerted only once every 30 days for an issue
- In the "IF" clause select "newer than 30 days" so that combined with the rate limit, you only get alerted once

The edge case here is that if an issue is first seen in an environment more than 30 days after it was created, you'll miss the alert.

**Comment by cpressland** [comment]
@adhiraj not sure I follow? We don't have an "IF" clause, unless I'm misunderstanding you?
<img width="1531" alt="Screenshot 2020-10-29 at 11 26 22" src="https://user-images.githubusercontent.com/6557665/97562520-a7dfed00-19d9-11eb-82e7-2458e33fe423.png">

**Comment by adhiraj** [comment]
@cpressland are you running self-hosted sentry? This is currently on the SaaS version, but we'll be releasing it to self-hosted in a few weeks.

**Comment by rachrwang** [comment]
Related issue: https://github.com/getsentry/sentry/issues/64354

**Comment by rachrwang** [comment]
Proposed solution: we add a new if condition for the "every event trigger" that says: "First time seen in {environment}", where {environment} is a dropdown
